src/python/pants/backend/jvm/subsystems/jvm_tool_mixin.py

```diff
@@ -152,11 +153,11 @@ def formulate_help():
   # TODO: Deprecate .register_jvm_tool()?
   class JvmToolDeclaration(datatype([
     ('tool_name', text_type),
-    'main',
+    'main',                     # Optional
```

Comment:
This could be expressed as `('main', (text_type, type(None))` I think?

What does a missing main mean? Worth documenting...